oneflow/api/python/framework/tensor_functions.cpp

```diff
@@ -273,6 +282,73 @@ static PyObject* PyTensorObject_reshape_as(PyObject* self, PyObject* args, PyObj
   END_HANDLE_ERRORS
 }
 
+static PyObject* PyTensorObject_view(PyObject* self, PyObject* args) {
+  HANDLE_ERRORS
+  PyObject* shape = args;
+  if (PyTuple_Size(args) == 1) {
+    PyObject* item = PyTuple_GetItem(args, 0);
+    if (!PyLong_Check(item)) { shape = item; }
+  }
+  CHECK_OR_THROW(functional::PyLongSequenceCheck(shape))
+      << Error::TypeError() << "view() received an invalid combination of arguments - got ("
+      << Py_TYPE(shape)->tp_name << "), but expected tuple of ints size.";
+
+  const auto& dims = functional::PyUnpackLongSequence<int64_t>(shape);
+  DimVector dim(dims.begin(), dims.end());
+  return PyTensor_New(ASSERT_PTR(functional::View(PyTensor_Unpack(self), Shape(dim))));
+  END_HANDLE_ERRORS
+}
+
+static PyObject* PyTensorObject_view_as(PyObject* self, PyObject* args, PyObject* kwargs) {
+  HANDLE_ERRORS
+  auto tensor = PyTensor_Unpack(self);
+  PyObject* other = NULL;
+  static const char* keywords[2] = {"other", NULL};
+  if (!PyArg_ParseTupleAndKeywords(args, kwargs, "O|:view_as", const_cast<char**>(keywords),
+                                   &other)) {
+    return NULL;
+  }
+  return PyTensor_New(ASSERT_PTR(functional::View(tensor, *PyTensor_Unpack(other)->shape())));
+  END_HANDLE_ERRORS
+}
+
+static PyObject* PyTensorObject_permute(PyObject* self, PyObject* args, PyObject* kwargs) {
+  HANDLE_ERRORS
+  PyObject* dims = args;
+  if (PyTuple_Size(args) == 0) {
+    // keyword parameter
+    static const char* keywords[2] = {"dims", NULL};
+    if (!PyArg_ParseTupleAndKeywords(args, kwargs, "O|:permute", const_cast<char**>(keywords),
+                                     &dims)) {
+      return NULL;
+    }
+  } else if (PyTuple_Size(args) == 1) {
+    // positional parameter
+    PyObject* item = PyTuple_GetItem(args, 0);
+    if (!PyLong_Check(item)) { dims = item; }
+  }
+  CHECK_OR_THROW(functional::PyLongSequenceCheck(dims))
+      << Error::TypeError() << "permute(): argument 'dims' must be tuple of ints, but found "
+      << Py_TYPE(dims)->tp_name;
+  const auto& dims_vec = functional::PyUnpackLongSequence<int32_t>(dims);
+  return PyTensor_New(ASSERT_PTR(functional::Permute(PyTensor_Unpack(self), dims_vec)));
+  END_HANDLE_ERRORS
+}
+
+static PyObject* PyTensorObject_transpose(PyObject* self, PyObject* args, PyObject* kwargs) {
+  HANDLE_ERRORS
+  auto tensor = PyTensor_Unpack(self);
+  int dim0 = 0;
```

Comment:
```suggestion
  PyObject* dim0 = PyLong_FromLong(0);
```
这里和dim1以及下面 `PyArg_ParseTupleAndKeywords` 的参数应该都是PyObject吧？这么写也可以正常运行吗？

Reply:
嗯可以运行的（应该不是PyObject，就是int吧😂

Reply:
学到了，我改一下我的写法